src/external/libpcre.m4

```diff
@@ -1,13 +1,45 @@
 AC_SUBST(PCRE_LIBS)
 AC_SUBST(PCRE_CFLAGS)
 
-PKG_CHECK_MODULES([PCRE], [libpcre], [found_libpcre=yes], [found_libpcre=no])
-PKG_CHECK_EXISTS(libpcre >= 7,
-                 [AC_MSG_NOTICE([PCRE version is 7 or higher])],
-                 [AC_MSG_NOTICE([PCRE version is below 7])
-                  AC_DEFINE([HAVE_LIBPCRE_LESSER_THAN_7],
-                            1,
-                            [Define if libpcre version is less than 7])])
+PKG_CHECK_MODULES(
+    [PCRE],
+    [libpcre],
+    [
+        found_libpcre=yes
+        PKG_CHECK_EXISTS(
+            libpcre >= 7,
+            [AC_MSG_NOTICE([PCRE version is 7 or higher])],
+            [
+                AC_MSG_NOTICE([PCRE version is below 7])
+                AC_DEFINE(
+                    [HAVE_LIBPCRE_LESSER_THAN_7],
+                    1,
+                    [Define if libpcre version is less than 7]
+                )
+            ]
+        )
+    ],
+    [
+        PKG_CHECK_MODULES(
+            [PCRE2],
+            [libpcre2-8],
+            [
+                found_libpcre=yes
+                AC_DEFINE(
+                    [HAVE_LIBPCRE2],
+                    1,
+                    [Define if libpcre2 is present]
+                )
+                AC_DEFINE(
+                    [PCRE2_CODE_UNIT_WIDTH],
+                    8,
```

Comment:
Where is this constant used? Since it's always defined to the same size, wouldn't it make more sense to just have a `#define` in the code?

Reply:
This constant is used in pcre2 and must be defined before include <pcre2>.

I put it here because it is connected with library variant we link, see few lines above [libpcre2-8] <- that 8, So we have this unit length in one place